src/test/java/com/wikia/webdriver/elements/mercury/pages/discussions/PostsListPage.java

```diff
@@ -91,14 +94,17 @@ public PostsListPage open() {
   }
 
   public PostsListPage waitForPageReload() {
+    waitSafely(() -> wait.forElementVisible(By.className("loading-overlay")));
+    waitSafely(() -> wait.forElementNotVisible(By.className("loading-overlay")));
+    return this;
+  }
+
+  private void waitSafely(Runnable o) {
     try {
-      wait.forElementVisible(By.className("loading-overlay"));
+      o.run();
     } catch (TimeoutException e) {
```

Comment:
Perhaps we can ask @ludwikkazmierczak to improve Sonar, so that it considers PageObjectLogging a logger.